src/views/ScenarioManager/ScenarioManager.js

```diff
@@ -47,69 +47,103 @@ const ScenarioManager = (props) => {
+  const getScenariolistAfterDelete = useCallback(
+    (idOfScenarioToDelete) => {
+      const scenarioListAfterDelete = scenarios
+        .map((scenario) => {
+          const newScenario = { ...scenario };
+          if (newScenario.parentId === idOfScenarioToDelete) {
+            newScenario.parentId = currentScenarioData.parentId;
+          }
+          return newScenario;
+        })
+        .filter((scenario) => scenario.id !== idOfScenarioToDelete);
+
+      return scenarioListAfterDelete;
+    },
+    [currentScenarioData?.parentId, scenarios]
+  );
+
+  const onScenarioDelete = useCallback(
+    (scenarioId) => {
+      const lastScenarioDelete = scenarios.length === 1;
+      deleteScenario(scenarioId);
+      if (scenarioId === currentScenarioData?.id) {
+        if (lastScenarioDelete) {
+          resetCurrentScenario();
+        } else {
+          setCurrentScenario(getFirstScenarioMaster(getScenariolistAfterDelete(scenarioId)));
         }
-        return newScenario;
-      })
-      .filter((scenario) => scenario.id !== idOfScenarioToDelete);
-
-    return scenarioListAfterDelete;
-  };
-
-  function onScenarioDelete(scenarioId) {
-    const lastScenarioDelete = scenarios.length === 1;
-    deleteScenario(scenarioId);
-    if (scenarioId === currentScenarioData.id) {
-      if (lastScenarioDelete) {
-        resetCurrentScenario();
-      } else {
-        setCurrentScenario(getFirstScenarioMaster(getScenariolistAfterDelete(scenarioId)));
       }
-    }
-  }
-
-  function onScenarioRename(scenarioId, newScenarioName) {
-    if (scenarioId === currentScenarioData.id) {
-      setCurrentScenario({ name: newScenarioName });
-    }
-    renameScenario(scenarioId, newScenarioName);
-  }
-
-  function checkScenarioNameValue(newScenarioName) {
-    const errorKey = ScenarioUtils.scenarioNameIsValid(newScenarioName, scenarios);
-    if (errorKey) {
-      const errorLabel = labels.scenarioRename.errors[errorKey];
-      if (!errorLabel) {
-        console.warn('Scenario error label key is broken !');
-        return 'Scenario name is invalid';
+    },
+    [
+      currentScenarioData?.id,
+      deleteScenario,
+      getScenariolistAfterDelete,
+      resetCurrentScenario,
+      scenarios.length,
+      setCurrentScenario,
+    ]
+  );
+
+  const onScenarioRename = useCallback(
+    (scenarioId, newScenarioName) => {
+      if (scenarioId === currentScenarioData?.id) {
+        setCurrentScenario({ name: newScenarioName });
+      }
+      renameScenario(scenarioId, newScenarioName);
+    },
+    [currentScenarioData?.id, renameScenario, setCurrentScenario]
+  );
+
+  const checkScenarioNameValue = useCallback(
+    (newScenarioName) => {
+      const errorKey = ScenarioUtils.scenarioNameIsValid(newScenarioName, scenarios);
+      if (errorKey) {
+        const errorLabel = labels.scenarioRename.errors[errorKey];
+        if (!errorLabel) {
+          console.warn('Scenario error label key is broken !');
```

Comment:
It was here before but can you delete the trailing space between _broken_ and exclamation mark ? :pray: :slightly_smiling_face:

Reply:
done